tests/test_utils.py

```diff
@@ -303,3 +306,437 @@ def check_multistream_classic(fnames):
+                                 sep='\t')
+
+            # remove space from column names in DataFrame
+            ebal_50 = ebal_50.rename(columns=lambda x: x.replace(' ', ''))
+
+            ebal_50['DATES'] = pd.to_datetime(ebal_50.YEAR * 10000 +
+                                         ebal_50.MONTH * 100 +
+                                         ebal_50.DAY, format='%Y%m%d')
+
+        except OSError:
+            print("this site has a lat/lng precision issue")
+            lat_issue=True
+            # To-Do: deal with lat/lng precision issue
+
+        # load VIC 5.0.x simulations
+        ebal_50x_file = 'en_bal_%s_%s.txt' %(lat, lng)
+
+        try:
+
+            # load VIC 5.x simulations
+            ebal_50x = pd.read_csv(os.path.join(result_dir,
+                                 ebal_50x_file),
+                                 skiprows=3,
+                                 sep='\t')
+
+            # remove space from column names in DataFrame
+            ebal_50x = ebal_50x.rename(columns=lambda x: x.replace(' ', ''))
+
+            ebal_50x['DATES'] = pd.to_datetime(ebal_50x.YEAR * 10000 +
+                                             ebal_50x.MONTH * 100 +
+                                             ebal_50x.DAY, format='%Y%m%d')
+
+
+        except OSError:
+            lat_issue=True
+            # To-Do: deal with lat/lng precision issue for some sites
+
+        # make figures
+
+        ecflux_vars = ['LE', 'H']
+        vic_vars = ['OUT_LATENT', 'OUT_SENSIBLE']
+        variable_names = ['Latent Heat', 'Sensible Heat']
+
+        # plot preferences
+        lw = 4.0
+        fs = 15
+        alpha = 1.0
+
+        if not lat_issue:
+            if 'annual_mean_diurnal_cycle' in plots_to_make:
+
+                # make annual mean diurnal cycle plots
+                f, axarr = plt.subplots(2, 1, figsize=(8,8))
+
+                for i, ecflux_var in enumerate(ecflux_vars):
+
+                    ebal_42.index = ebal_42.DATES
+
+                    # plt VIC 4.2
+                    axarr[i].plot(ebal_42[vic_vars[i]].groupby(ebal_42['HOUR']).mean(),
+                                'b', label='VIC 4.2', linewidth=lw)
+
+                    # plot VIC 5.0
+                    ebal_50.index = ebal_50.DATES
+
+                    # convert seconds column to hours for groupby
+                    ebal_50['HOUR'] = (ebal_50['SEC'] * (1/3600)).astype(int)
+
+                    axarr[i].plot(ebal_50[vic_vars[i]].groupby(ebal_50['HOUR']).mean(),
+                                'r', label='VIC 5.0', linewidth=lw)
+
+                    # plot VIC 5.0.x
+                    ebal_50x.index = ebal_50x.DATES
+
+                    # convert seconds column to hours for groupby
+                    ebal_50x['HOUR'] = (ebal_50x['SEC'] * (1/3600)).astype(int)
+
+                    axarr[i].plot(ebal_50x[vic_vars[i]].groupby(ebal_50x['HOUR']).mean(),
+                                'y', label='VIC 5.0.x', linewidth=lw)
```

Comment:
There is a bunch of repeated code in these lines (626:648).  A more extensible approach would be something like this:

*Psudo code...*
```Python

datasets = '''dict of datasets, all formatted identically'''

for key, dset in datasets.items():
    if varname in dset:
        dset[varname].groupby(lambda x: x.hour).mean().plot(ax=axarr[i], 'b', label=key, linewidth=lw)
```

You'll need to make sure all the datasets are formatted identically before you come into the plot step.  This means:
- All have datetime indexes (this will save you from needing to use the `'SEC'` columns.
- All all aligned (maybe easiest to just put them in the same dataframe)
- All have the same variable names (you may need to use `rename` in some cases)